x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/start_datafeed_modal/time_range_selector/time_range_selector.js

```diff
@@ -216,6 +214,48 @@ function TabStack({ title, items, switchState, switchFunc }) {
     </div>
   );
 }
+
+const DatePickerWithInput = ({ date, onChange, minDate, setIsValid }) => {
+  const [dateString, setDateString] = useState(date.format(TIME_FORMAT));
+
+  function onTextChange(e) {
+    const val = e.target.value;
+    setDateString(val);
+
+    const newDate = moment(val);
+    if (newDate.isValid()) {
+      setIsValid(true);
+      onChange(newDate);
+    } else {
+      setIsValid(false);
+    }
+  }
+
+  function onCalendarChange(newDate) {
+    setDateString(newDate.format(TIME_FORMAT));
+    setIsValid(true);
+    onChange(newDate);
+  }
+
+  return (
+    <>
+      <EuiFieldText
+        value={dateString}
+        onChange={onTextChange}
+        placeholder={TIME_FORMAT}
+        aria-label="CHANGE ME"
+      />
+      <EuiDatePicker
```

Comment:
If you enter a date/time manually, then click one of the other options, such as 'Start from beginning of data', then reselect 'Specify start time', the previous text is still displayed but the calendar has gone back to 1970. Could the calendar be updated when switching back to 'Specify start time', or just reset the text?

![image](https://user-images.githubusercontent.com/7405507/72913679-6b3de180-3d35-11ea-980b-f076e7d25d3f.png)

Reply:
addressed in 4f440fdc1d00aa6f89201b438473779441a3f5c8
When switching between the type of start time, it will be difficult to remember the previous user entered date.
However there was definitely a bug here, seen in this screenshot, where the calendar and text dates can become different when switching the types of start times.